### ISSUE TITLE
Is this 5.7 compatible?

### ISSUE THREAD

**Issue by isbkch** [stechstudio/laravel-metrics#4]
I'm sending metrics to cloudwatch from Laravel 5.7 but I never seen them.

Also trying to implement in an Event class adding `implements ShouldReportMetric {` gives a php error about the class must be to be abstract.

is this package compatible with Laravel 5.7?

**Comment by jszobody** [comment]
We are using this with Laravel 5.7, yes.

Implementing an interface shouldn't be related to an error about a class needing to be abstract, I'm going to guess you have a different issue going on there. Maybe post your whole Event class so I can take a look?

**Comment by isbkch** [comment]
Sure, here is my event class.

```<?php

namespace App\Events;

use App\Models\Post;
use Illuminate\Queue\SerializesModels;
use Illuminate\Foundation\Events\Dispatchable;
use Illuminate\Broadcasting\InteractsWithSockets;

class PostSaved
{
    use Dispatchable, InteractsWithSockets, SerializesModels;

    public $post;
    /**
     * Create a new event instance.
     *
     * @return void
     */
    public function __construct(Post $post)
    {
        $this->post = $post;
    }

    /**
     * Get the channels the event should broadcast on.
     *
     * @return \Illuminate\Broadcasting\Channel|array
     */
    public function broadcastOn()
    {
        return [];
    }
}
```

That I change to be:
```
<?php

namespace App\Events;

use App\Models\Post;
use Illuminate\Queue\SerializesModels;
use STS\Metrics\Contracts\ShouldReportMetric;
use Illuminate\Foundation\Events\Dispatchable;
use Illuminate\Broadcasting\InteractsWithSockets;

class PostSaved implements ShouldReportMetric
{
    use Dispatchable, InteractsWithSockets, SerializesModels;

    public $post;
  // rest is same
```
which result in the following error
```message: "Class App\Events\PostSaved contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (STS\Metrics\Contracts\ShouldReportMetric::createMetric)"
```

**Comment by jszobody** [comment]
Ok it's looking for the `createMetric` method, which the interface requires. You need to use the trait that builds this for you, or build it yourself.

See here:

https://github.com/stechstudio/laravel-metrics#1-use-the-providesmetric-trait

You probably just want to the use trait, that will send a `post_saved` metric with a value of `1` by default. You can customize that further if needed.

**Comment by isbkch** [comment]
No errors anymore but still not seeing anything on Cloud watch Metric. I assume the error must be somewhere else. Thanks for your help

**Comment by jszobody** [comment]
We primarily use InfluxDB as our backend, haven't tried CloudWatch in a while.

So I just spun up a fresh Laravel 5.7 app, added this package along with `aws-sdk-php-laravel`. I created a `PostSaved` event class (simpler than yours, no Post model, and no sockets/broadcasting going on). 

I created a new IAM user in AWS, assigned it the `CloudWatch-PutMetric` policy so it had sufficient permission. Put the credentials in my .env file, and also put this:

```
METRICS_BACKEND=cloudwatch
CLOUDWATCH_NAMESPACE=testing
```

I then modified the routes file:

```
Route::get('/', function () {
    event(new \App\PostSaved());
    return view('welcome');
});
```

And loaded the Laravel welcome screen. Then I checked CloudWatch:

![image](https://user-images.githubusercontent.com/203749/53246793-a0eace00-367f-11e9-9467-119984d73fe3.png)

Namespace is there, and the `post_saved` metric is included.

All that's to say... I can now be confident the package works well with Laravel 5.7 and CloudWatch specifically!

You probably want to review your AWS user IAM permissions, make sure it has a policy that allows it to send metrics to CloudWatch. You might also consider removing the socket/broadcasting stuff temporarily, just to troubleshoot. I don't _think_ that should impact this package at all, but worth verifying (and let me know if that's the issue!)
